text/0000-image-retention-policy.md

```diff
@@ -0,0 +1,52 @@
+# Define an Image Retention Policy for Paketo Images
+
+## Summary
+
+At the moment, the Paketo Project is maintaining images (buildpack, builder, stack, etc..) since the beginning of the project. This RFC proposes that we define an image retention policy that would allow the project to delete old images.
+
+## Motivation
+
+Multiple reasons:
+
+1. It is impractical from a cost and resources perspective for the project to maintain all images throughout the history of time.
+2. It is highly unlikely that images going back to the beginning of the project are still usable. Most buildpacks reference external dependencies and those dependencies may no longer exist.
+3. Older stacks, buildpacks and builder images very likely are full of vulnerable software and should no longer be used.
```

Comment:
These images certainly contain vulnerabilities and shouldn't be used, but I don't see that as a strong argument for pulling them. I think having an immutable history of release artifacts is generally in the best interest of users and we should strive to maintain that unless it becomes impossible for cost or logistical reasons.

Reply:
There will still be release artifacts going back to the first release. The primary release artifact is the source code, which is tagged and exists forever on Github (at least as long as Github is around). 

There would also continue to be an archived history in the buildpacks registry. I talked with that team and if we use the "yank" functionality for old releases, that would retain them in the registry but indicate that it's no longer recommended that users consume them. I think this is exactly the statement that we want to make for old and vulnerable images.

In addition, if users need to retain those images for longer periods of time they can easily do so. It's a matter of `crane cp` (or insert your tool of choice) to move the required images onto the user's own storage. The benefit of having a retention policy in place is that users can plan around it. They'll know how the project will maintain images & they can make arrangements to store them elsewhere if that's a requirement they need.

>I think having an immutable history of release artifacts is generally in the best interest of users and we should strive to maintain that unless it becomes impossible for cost or logistical reasons.

I guess I would flip this around and say, why should we keep them? Do we really think anyone is going to use them? How frequently do we think someone is actually going to take advantage of us retaining them?

It's not just the overall cost, but the cost (and cost is not just financial here, I would include environmental cost as well)/benefit. Just because it's possible, should we accept a cost because one day, someone might come around and use one of the images? I think the policy can also help to fairly draw a line in terms of what is an acceptable cost/benefit ratio for the project.

If we pick a cut-off point where 99% (probably even higher, 99.9% or 99.99%) are not using these images, then what is the harm of reclaiming them? To me, the likelihood of someone needing Paketo Buildpacks Java v1.0.0 is very close to zero, unless someone decided to perform a software archaeology or research study project.